dips/dip-158.md

```diff
@@ -0,0 +1,795 @@
+|
+
+#### VoidCommand Request
+Field|Type|Required?|Description|
+|-|-|-|-|
+|reference_id|str|Y|Unique reference ID of this payment. This value should be globaly unique|
+|status|StatusObject|Y|Receiver status. For the void command the status should be abort and the abort code should be void|
+|
+```
+{
+  "_ObjectType": "CommandRequestObject",
+  "command_type": "VoidCommand",
+  "cid": "3185027f-0574-6f55-2668-3a38fdb5de98",
+  "command": {
+    "_ObjectType": "VoidCommand",
+    "reference_id": "4185027f-0574-6f55-2668-3a38fdb5de98",
+    "status": {
+      "status": "abort",
+      "abort_code": "void",
+      "abort_message": "abort message"
+    }
+  }
+}
+
+
+
+
+```
+#### VoidCommand Response
+Field|Type|Required?|Description|
+|-|-|-|-|
+|status|StatusObject|Y|Sender status. Used to verify that the sender VASP acknoledges the void command by responding with the abort status|
+|
+```
+{
+  "_ObjectType": "CommandResponseObject",
+  "status": "success",
+  "cid": "12ce83f6-6d18-0d6e-08b6-c00fdbbf085a",
+  "result": {
+    "_ObjectType": "StatusObject",
+    "status": {
+      "status": "abort"
+    }
+  }
+}
+```
+
+### AbortCommand
+
+| Functionality |Direction| Description | Command Type | Request Data | Response Data |
+|-|-|-|-|-|-|
+|ABORT_PAYMENT|Sender > Receiver / Receiver > Sender|This functionality allows both parties to abort the payment. For example, the receiver may want to abort the payment following failed risk checks|AbortCommand|Reference ID ; Requester (Sender / Receiver) Status|Responder (Sender / Receiver) Status|
+|
+
+#### AbortCommand Request
+Field|Type|Required?|Description|
+|-|-|-|-|
+|reference_id|str|Y|Unique reference ID of this payment. This value should be globaly unique|
+|sender / receiver|PaymentActorObject|Y|Indicates the initiator of the abort command (can be either the sender or the receiver). Must contain a nested StatusObject (other elements are not required)|
+|
+```
+{
+  "_ObjectType": "CommandRequestObject",
+  "command_type": "AbortCommand",
+  "cid": "3185027f-0574-6f55-2668-3a38fdb5de98",
+  "command": {
+    "_ObjectType": "AbortCommand",    
+    "reference_id": "4185027f-0574-6f55-2668-3a38fdb5de98",
+    "sender": {
+      "status": {
+        "status":"abort",
+        "abort_code": "rejected",
+        "abort_message": "abort message"
+      }
+    }
+  }
+}
+
+```
+#### AbortCommand Response
+Field|Type|Required?|Description|
+|-|-|-|-|
+|status|StatusObject|Y|Responder status. Used to verify that the responding VASP acknoledges the abort command by responding with the abort status|
+|
+```
+{
+  "_ObjectType": "CommandResponseObject",
+  "status": "success",
+  "cid": "12ce83f6-6d18-0d6e-08b6-c00fdbbf085a",
+  "result": {
+    "_ObjectType": "StatusObject",
+    "status": {
```

Comment:
maybe just the status enum?

Reply:
I think we gain from the consistency here (easier implementation).